src/sdk/pynni/nni/compression/tensorflow/compressor.py

```diff
@@ -1,18 +1,20 @@
 import logging
 import tensorflow as tf
 from . import default_layers
+tf.config.experimental_run_functions_eagerly(True)
 
 _logger = logging.getLogger(__name__)
 
 
 class LayerInfo:
-    def __init__(self, op, weight, weight_op):
-        self.op = op
-        self.name = op.name
-        self.type = op.type
-        self.weight = weight
-        self.weight_op = weight_op
-
+    def __init__(self, keras_layer):
+        self.keras_layer = keras_layer
+        self.name = keras_layer.name
+        self.type = default_layers.get_op_type(type(keras_layer))
+        self.weight_index = default_layers.get_weight_index(self.type)
```

Comment:
there is only one weight or could be multiple weights (thus indexes)?

Reply:
There could be multiple weights, such as:  https://github.com/tensorflow/tensorflow/blob/befd063e093b306084642872c1c7e95317e0fee4/tensorflow/python/keras/layers/convolutional.py#L1569

For most layers with weight, there is a weight and a bias,  the index of weight is normally 0, index of bias is normally 1.